dali/imgcodec/decoders/libtiff/tiff_libtiff.cc

```diff
@@ -223,21 +277,40 @@ DecodeResult LibTiffDecoderInstance::Decode(SampleView<CPUBackend> out, ImageSou
   TensorShape<> out_line_shape = {roi.shape()[1], out_channels};
   TensorShape<> out_line_strides = kernels::GetStrides(out_line_shape);
 
+  // We choose smallest possible type
+  size_t in_type_bits;
+  if (info.bit_depth <= 8) in_type_bits = 8;
+  else if (info.bit_depth <= 16) in_type_bits = 16;
+  else if (info.bit_depth <= 32) in_type_bits = 32;
+  else
+    DALI_FAIL(make_string("Unsupported bit depth: ", info.bit_depth));
```

Comment:
doesn't this duplicate the error check in line 167?
If this code path should never be reached maybe you can change to an assert.

Reply:
Yes, it should never get here. I'll add an assert in a moment